### ISSUE TITLE
Does the batch_size must be set to 2?

### ISSUE THREAD

**Issue by xiehousen** [ZwwWayne/K-Net#12]
`        
if cls_scores is None:
            detached_cls_scores = [None] * 2
        else:
            detached_cls_scores = cls_scores.detach()
for i in range(num_imgs):
          assign_result = self.assigner.assign(scaled_mask_preds[i].detach(),
                                                 detached_cls_scores[i],
                                                 gt_masks[i], gt_labels[i],
                                                 img_metas[i])`

as can be seen from above code, 
`len(detached_cls_scores)=2`

**Comment by ZwwWayne** [comment]
It seems that this is a hard code, I will fix it later. Thank you~

**Comment by ZwwWayne** [comment]
resolved in https://github.com/ZwwWayne/K-Net/commit/5e50ee58957dce972f51096804ff69171c2f072e

**Comment by AlphaNext** [comment]
> resolved in [5e50ee5](https://github.com/ZwwWayne/K-Net/commit/5e50ee58957dce972f51096804ff69171c2f072e)

@ZwwWayne  If I set batch size to 8, the same problem may occur in Instance Segmentation task with this [config file](https://github.com/ZwwWayne/K-Net/blob/main/configs/det/knet/knet_s3_r50_fpn_ms-3x_coco.py).

 Is there problem in file [kernel_head.py](https://github.com/ZwwWayne/K-Net/blob/main/knet/det/kernel_head.py#L480-L481)  and [kernel_update_head.py](https://github.com/ZwwWayne/K-Net/blob/main/knet/det/kernel_update_head.py#L421-L422)?
